service/api_logic/user_logic.py

```diff
@@ -67,59 +74,49 @@ def __send_reset_email(self, user: User, token: str):
 
 
     def __get_reset_token(self, email) -> str:
-        user = self._user_dal.get_user_by_email_or_username(email)
+        user = self.get_user_by_email_or_username(email)
         if not user:
             raise UserDoesNotExistError(email)
 
         return self._serializer.dumps(user.username, salt = "email-confirm")
 
 
-    def reset_user_password(self, email, new_password: str) -> str:
-        user = self._user_dal.get_user_by_email_or_username(email)
+    def reset_user_password(self, email, new_password: str):
+        user = self.get_user_by_email_or_username(email)
         if not user:
             raise UserDoesNotExistError(email)
 
         self._user_dal.update_user_password(user, new_password)
-        new_jwt = create_access_token(identity = user.email)
+        new_jwt = create_access_token(identity=user.email)
+        new_refresh = create_refresh_token(identity=user.email)
 
-        return new_jwt
+        response = jsonify({"message": "Password reset successful"})
+        set_access_cookies(response, new_jwt)
+        set_refresh_cookies(response, new_refresh)
+
+        return response
 
 
     def confirm_token(self, token: str, expiration=3600):
         username = self._serializer.loads(token, salt = "email-confirm", max_age = expiration)
-        user = self._user_dal.get_user_by_email_or_username(None, username)
+        user = self.get_user_by_email_or_username(None, username)
 
         return OutputUser().dump(user)
 
 
-    async def log_in(self, email_or_username: str, password: str):
-        user = self._user_dal.get_user_by_email_or_username(email_or_username, email_or_username)
-
-        if not user:
-            self._logger.warning("User does not exist")
-            raise IncorrectUsernameOrEmailError()
-
-        if not bcrypt.checkpw(password.encode('utf-8'), user.password_hash.encode('utf-8')):
-            self._logger.warning("Passwords do not match")
-            raise IncorrectPasswordError()
-
-        token = await self.__generate_auth_token(user)
-
-        return OutputLogin(email = user.email, token = token, id = user.user_id)
+    async def log_in(self, credentials: InputUserLogInDTO):
+        login_context = AuthManager(self)
+        user = await login_context.execute_log_in(credentials.auth_provider, credentials)
```

Comment:
MINOR:

`credentials.auth_provider, credentials` doesn't sound nice does it?
We either just pass "credentials" (whole object), or do some proper inheritance, like:

`class InputUserDto:
    auth_provider: AuthMethods (your auth methods Enum)
    user_credentials: UserCredentials (your dto for creds)
`

then it'll be like: user = await login_context.execute_log_in(nameofdto.auth_provider, nameofdto.user_credentials)